### DISCUSSION TITLE
v0.54 error on make

### DISCUSSION THREAD

**Discussion by rkarikari** [jvde-github/AIS-catcher#192] [General]
Hello,
I got the following error below when trying to update to v0.54 .......
git pull  and cmake ..  werew without any issues.

any ideas what I can do ?

`➜  build git:(main) ✗ make    
Consolidate compiler generated dependencies of target AIS-catcher
[  2%] Building CXX object CMakeFiles/AIS-catcher.dir/IO/HTTPClient.cpp.o
In file included from /Users/K/compiled/ais-catcher/IO/HTTPClient.cpp:20:
/Users/K/compiled/ais-catcher/./IO/HTTPClient.h:28:10: warning: non-portable path to file '"zip.h"'; specified path differs in case from file name on disk [-Wnonportable-include-path]
#include "ZIP.h"
         ^~~~~~~
         "zip.h"
In file included from /Users/K/compiled/ais-catcher/IO/HTTPClient.cpp:20:
In file included from /Users/K/compiled/ais-catcher/./IO/HTTPClient.h:29:
/Users/RichardK/compiled/ais-catcher/./Library/TCP.h:63:21: error: implicit instantiation of undefined template 'std::__1::vector<char, std::__1::allocator<char> >'
                std::vector<char> out;
                                  ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iosfwd:216:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector;
                           ^
In file included from /Users/K/compiled/ais-catcher/IO/HTTPClient.cpp:20:
/Users/K/compiled/ais-catcher/./IO/HTTPClient.h:40:3: error: unknown type name 'ZIP'
                ZIP zip;
                ^
1 warning and 2 errors generated.
make[2]: *** [CMakeFiles/AIS-catcher.dir/IO/HTTPClient.cpp.o] Error 1
make[1]: *** [CMakeFiles/AIS-catcher.dir/all] Error 2
make: *** [all] Error 2`

**Comment by jvde-github** [comment]
Can you provide more details on the system on which you are trying to build?

**Comment by jvde-github** [comment]
I uploaded a fix that should help for macos builds. Can you retry?

**Comment by rkarikari** [comment]
Thanks so much... worked just fine with RTLSDR  V3

but nothing with V4.

same behaviour with the previous  v.53

Have I missed something ?




<img width="1027" alt="Screenshot 2023-10-30 at 7 55 01 PM" src="https://github.com/jvde-github/AIS-catcher/assets/115777368/865d7d9f-c6d4-405d-8df0-4703b459a360">


<img width="1063" alt="Screenshot 2023-10-30 at 7 57 30 PM" src="https://github.com/jvde-github/AIS-catcher/assets/115777368/ab86063f-4e76-497a-b32b-ea853887aabc">

**Comment by jvde-github** [comment]
Cool (on the build process fix).

This is not an AIS-catcher issue for the V4 as it simply uses the library that is installed on the system. This [link](https://www.rtl-sdr.com/v4/) has a recipe under the header "Compiling RTL-SDR on MacOS" . You might have to rebuild AIS-catcher after that.

ps I don't have a V4 yet so cannot test.

**Comment by rkarikari** [comment]
This is interesting......
RTLSDR compiled and updated to latest version for V4 support.

V3 still works great with all software tested.


For V4, 

Gqrx works  (normal tuning through entire supported frequency range, no DIrect sampling required)
SDR++ works  (normal tuning through entire supported frequency range, no DIrect sampling required)
SDRAngel works (normal tuning through entire supported frequency range, no DIrect sampling required)  AIS works in SDRangel !!


AIS-catcher does not work
RTL_FM does not work ....
 .... with rtlfm, found a cluster of commercial broadcast FM stations approx -30Mhz off-frequency 
....53Mhz and above instead of ( 88 - 108 )Mhz

**Reply by jvde-github** [reply]
The packages you mention come distributed with a build version of the rtl-sdr library. AIS-catcher and rtl-fm rely on the installed system library. Possibly there is still an old/other version on the system?

This line from SDR++ build worked on my mac: 
```
git clone https://github.com/osmocom/rtl-sdr && cd rtl-sdr && mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=Release .. && make -j3 LIBRARY_PATH=$(pkg-config --libs-only-L libusb-1.0 | sed 's/\-L//') && sudo make install && cd ../../
```
Then if you rebuild AIS-catcher, cmake shows the directory where it finds the rtl-sdr library, I would check that this is the same as the one installed.

I hopefully receive the V4 in two weeks, will have a test then as well.

**Comment by rkarikari** [comment]
Here's the problem.....
The files are bein installed to ...       /usr/local      instead of          /opt/local


```
-- Installing: /usr/local/lib/librtlsdr.0.6git.dylib

-- Up-to-date: /usr/local/lib/librtlsdr.0.dylib
-- Up-to-date: /usr/local/lib/librtlsdr.dylib
-- Installing: /usr/local/lib/librtlsdr.a
-- Installing: /usr/local/bin/rtl_sdr
-- Installing: /usr/local/bin/rtl_tcp
-- Installing: /usr/local/bin/rtl_test
-- Installing: /usr/local/bin/rtl_fm
-- Installing: /usr/local/bin/rtl_eeprom
-- Installing: /usr/local/bin/rtl_adsb
-- Installing: /usr/local/bin/rtl_power

-- Installing: /usr/local/bin/rtl_biast
```


AIS-catcher is locating the files at ...

`-- RTLSDR: found - /opt/local/include, /opt/local/lib/librtlsdr.dylib`


why the change from /opt/local    to    /usr/local    ?


any ideas how to get this back to normal behaviour ?

**Reply by jvde-github** [reply]
Have you uninstalled any previous install of lib rtl sdr? e.g. as installed with brew and then after that install rtl-sdr via the manual way to get the latest code?

**Comment by rkarikari** [comment]
Yes, while troubleshooting I uninstalled all rtlsdr instances I could find.

The file ` librtlsdr.dylib`  does not exist at `/opt/local `       but ais-catcher is still claiming to have located it there.

**Comment by rkarikari** [comment]
After manually deleting all rtlsdr refs.  at     /opt/local,  all works now.

<img width="1018" alt="Screenshot 2023-10-31 at 10 08 52 PM" src="https://github.com/jvde-github/AIS-catcher/assets/115777368/2768fa73-bde3-4c99-8c4e-7ea81dcabc1f">




```
-- RTLSDR: found - /opt/local/include,     /usr/local/lib/librtlsdr.dylib
```

Thank you so much.

Has support for PostgreSQL been removed ?
just noticed.

**Comment by jvde-github** [comment]
Great! I think I need to add something so that users can manually overwrite certain directories to fix these problems PSQL is supported on Mac but CMAKE cannot find the brew directories for it. Not sure how to fix this except for adding an option so users can set on the command line.
